src/dc_receive_imf.rs

```diff
@@ -257,7 +257,7 @@ pub unsafe fn dc_receive_imf(
         );
     }
 
-    if !mime_parser.message_kml.is_none() && chat_id > DC_CHAT_ID_LAST_SPECIAL {
+    if mime_parser.location_kml.is_some() || mime_parser.message_kml.is_some() {
```

Comment:
this is a bugfix.

Reply:
What happened to `chat_id` check, why was it there and why it is not necessary anymore?

Reply:
the check is on the first statement of save_locations()